### ISSUE TITLE
`no-unsupported-features` allows unsupported Array prototype methods on Node runtime version v4.x.x

### ISSUE THREAD

**Issue by pdulapalli** [mysticatea/eslint-plugin-node#121]
I am running Node.js version v4.2.2 with the `no-unsupported-features` ruleset enabled. I use the `engines` field in my `package.json` as follows:

```
  "engines": {
    "node": "4.2.2"
  }
```

 [According to the compatibility matrix](https://node.green/#ES2016-features-Array-prototype-includes), `Array.prototype.includes` is unsupported.

Also, `Array.prototype.values` seems to also be missed by the `no-unsupported-features` rules.

---

Here is a sample of my `.eslintrc.json` file:

```
"plugins": ["chai-friendly", "node", "promise"],
"extends": ["airbnb-base", "plugin:promise/recommended", "plugin:node/recommended"],
"env": {
    "es6": true,
    "node": true,
    "mocha": true
},
"parserOptions": {
    "ecmaVersion": 2018
}
```

I don't think that the other plugins and env settings override `node/recommended` (`no-unsupported-features`). As far as I can tell, all the other unsupported ES6+ features I attempt to use on my runtime are detected and warned against just fine.

**Comment by mysticatea** [comment]
Thank you for this issue.

Unfortunately, that rule doesn't support prototype properties. The documentation mentions it as a known limitation: https://github.com/mysticatea/eslint-plugin-node/blob/e957849ce0691794741fb793cab0e61090773340/docs/rules/no-unsupported-features.md#known-limitations

Because ESLint is a static analyzer which checks each file. It cannot know the type of variables without type annotations.
